dataset/diarization.py

```diff
@@ -0,0 +1,59 @@
+import os
+import json
+import tqdm
+import argparse
+import vad
+import audio
+import functools
+import transcripts
+import multiprocessing as mp
+
+
+def make_diarization_dataset(input_path: str, output_path: str, sample_rate: int, keep_intersections: bool, vad_type: str, device: str, processes: int):
+	if os.path.isdir(input_path):
+		audio_files = [os.path.join(input_path, audio_name) for audio_name in os.listdir(input_path)]
+	else:
+		audio_files = [input_path]
+
+	if vad_type == 'simple':
+		_vad = vad.PrimitiveVAD(device = device)
+	elif vad_type == 'webrtc':
+		_vad = vad.WebrtcVAD(sample_rate = sample_rate)
+	else:
+		raise RuntimeError(f'VAD for type {vad_type} not found.')
+
+	if processes > 0:
+		parametrized_generate = functools.partial(generate_markup, sample_rate=sample_rate, vad=_vad, keep_intersections=keep_intersections)
+		with mp.Pool(processes=processes) as pool:
+			dataset = list(tqdm.tqdm(pool.imap(parametrized_generate, audio_files), total=len(audio_files)))
+	else:
+		dataset = [generate_markup(audio_path, sample_rate, _vad, keep_intersections) for audio_path in tqdm.tqdm(audio_files)]
+
+	with open(output_path, 'w') as output:
+		output.write('\n'.join([json.dumps(example, ensure_ascii = False) for example in dataset]))
+
+
+def generate_markup(audio_path: str, sample_rate: int, vad, keep_intersections: bool):
```

Comment:
what is semantics of `markup`? what's data structure representing it?

Reply:
`markup` -> `intervals`